mlflow/pytorch/__init__.py

```
@@ -1099,3 +1125,79 @@ def print_auto_logged_info(r):
+        global_step: The global step of the checkpoint to be loaded, if
+          you set "checkpoint_save_freq" to an integer.
+    """
+    from mlflow.pytorch._lightning_autolog import LATEST_CHECKPOINT_ARTIFACT_TAG_KEY
+
+    client = MlflowClient()
+
+    if run_id is None:
+        run = mlflow.active_run()
+        if run is None:
+            raise MlflowException(
+                "There is no active run, please provide the 'run_id' for " "'load_checkpoint' call."
+            )
+        run_id = run.info.run_id
+    else:
+        run = client.get_run(run_id)
+
+    latest_checkpoint_artifact_path = run.data.tags.get(LATEST_CHECKPOINT_ARTIFACT_TAG_KEY)
+    if latest_checkpoint_artifact_path is None:
+        raise MlflowException("There is no logged checkpoint artifact in current run.")
+
+    checkpoint_filename = os.path.basename(latest_checkpoint_artifact_path)
+
+    if epoch is not None and global_step is not None:
+        raise MlflowException(
+            "You can only set one of 'epoch' and 'global_step' for 'load_checkpoint'."
+        )
+    elif global_step is not None:
+        checkpoint_artifact_path = f"checkpoints/global_step_{global_step}/{checkpoint_filename}"
+    elif epoch is not None:
+        checkpoint_artifact_path = f"checkpoints/epoch_{epoch}/{checkpoint_filename}"
+    else:
+        checkpoint_artifact_path = latest_checkpoint_artifact_path
+
+    downloaded_checkpoint_filepath = client.download_artifacts(run_id, checkpoint_artifact_path)
+    model = model_class.load_from_checkpoint(downloaded_checkpoint_filepath)
+    return model
+
+
+__all__ = [
+    "autolog",
+    "load_model",
+    "save_model",
+    "log_model",
+    "get_default_pip_requirements",
+    "get_default_conda_env",
+]
+
+try:
+    from mlflow.pytorch._lightning_autolog import MLflowModelCheckpointCallback
+    __all__.append("MLflowModelCheckpointCallback")
+except ImportError:
+    # Swallow exception if pytorch-lightning is not installed.
```

I think we can skip this try catch block because we are using lazy loading.

---

Seemingly lazy loading doesn't address it.

```
lightning_autolog = LazyLoader("mlflow. pytorch. _lightning_autolog", globals(), "mlflow. pytorch. _lightning_autolog")  # no exception
lightning_autolog.MLflowModelCheckpointCallback  # exception is raised if pytorch-lightning is not installed.
```

---

and if we don't swallow the exception here,

when initializaing the `mlflow.pytorch` module (even if it is lazy-loading, it needs initialization when loaded), exception is raised.

---

you are right, I forgot that this is called pytorch module. In fact this autologging part should go to`lightning` or `pytorch_lightning`, but it's independent of this PR.